tests_integration/test_snowpark_function.py

```diff
@@ -0,0 +1,417 @@
+        function_name = (
+            self._setup.test_object_name_provider.create_and_get_next_object_name()
+        )
+        result = self._setup.runner.invoke_with_config_and_integration_connection(
+            [
+                "snowpark",
+                "function",
+                "create",
+                "--name",
+                function_name,
+                "--handler",
+                "app.hello",
+                "--input-parameters",
+                "()",
+                "--return-type",
+                "string",
+            ]
+        )
+        assert_that_result_is_successful(result)
+        return function_name
+
+    def snowpark_function_update_should_finish_successfully(
+        self,
+        function_name: str,
+    ) -> None:
+        replace_text_in_file(
+            file_path="app.py",
+            to_replace="def hello() -> str:",
+            replacement="def hello() -> int:",
+        )
+        replace_text_in_file(
+            file_path="app.py",
+            to_replace='return "Hello World!"',
+            replacement="return 1",
+        )
+        result = self._setup.runner.invoke_with_config_and_integration_connection(
+            [
+                "snowpark",
+                "function",
+                "update",
+                "--name",
+                function_name,
+                "--handler",
+                "app.hello",
+                "--input-parameters",
+                "()",
+                "--return-type",
+                "int",
+            ]
+        )
+        assert_that_result_is_successful(result)
+
+    def snowpark_function_drop_should_finish_successfully(
+        self,
+        function_name: str,
+        arguments: str,
+    ) -> None:
+        result = self._setup.runner.invoke_with_config_and_integration_connection(
+            [
+                "snowpark",
+                "function",
+                "drop",
+                "--name",
+                function_name,
+                "--input-parameters",
+                arguments,
+            ]
+        )
+        assert_that_result_is_successful(result)
+
+    def assert_that_no_functions_are_in_snowflake(self) -> None:
+        self.assert_that_only_these_functions_are_in_snowflake()
+
+    def assert_that_only_these_functions_are_in_snowflake(
+        self, *expected_full_function_signatures: str
+    ) -> None:
+        actual_function_signatures = [
+            function["arguments"]
+            for function in self._setup.query_functions_created_in_this_test_case()
+        ]
+        adjusted_expected_full_function_signatures = [
+            function.upper() for function in expected_full_function_signatures
+        ]
+        assert set(actual_function_signatures) == set(
+            adjusted_expected_full_function_signatures
+        )
+
+    def assert_that_no_files_are_staged_in_snowflake(self) -> None:
+        self.assert_that_only_these_files_are_staged_in_snowflake()
```

Comment:
Alias for method? 🤔

Reply:
yes, to reflect in method name what exactly I want to assert